var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -329,13 +340,16 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
 
     conflicts("~openmp", when="@8:", msg="Building without OpenMP is not supported in CP2K 8+")
 
+    if (not when("+cuda")) or (not when("+rocm")):
+        conflicts("spla_gemm_offloading")
```

Comment:
Does `cp2k` build with this syntax? This should probably be:
```suggestion
    conflicts("+spla_gemm_offloading", "~rocm ~cuda")
```
?

Sorry I missed this the first time around.

Reply:
This suggestion got reverted.

Reply:
You are right it does not work. There is a better way to do it using variant or the solution offered.